### ISSUE TITLE
Error when getting a callback from Dr Chrono

### ISSUE THREAD

**Issue by ShepBook** [johnnadeau/omniauth-drchrono#2]
Here's what I've been getting.

```
Started GET "/users/auth/drchrono" for 127.0.0.1 at 2016-08-03 17:38:25 -0600
I, [2016-08-03T17:38:25.750204 #3077]  INFO -- omniauth: (drchrono) Request phase initiated.


Started GET "/users/auth/drchrono/callback?state=9d2a54295090a659aa20c695e6a9d75005596963efeb27bc&code=ckkqWoTXmXRcC4GIoMvwekRbrZt9qU" for 127.0.0.1 at 2016-08-03 17:38:40 -0600
I, [2016-08-03T17:38:40.879915 #3077]  INFO -- omniauth: (drchrono) Callback phase initiated.
E, [2016-08-03T17:38:41.584783 #3077] ERROR -- omniauth: (drchrono) Authentication failure! invalid_credentials: OAuth2::Error, access_denied:
{"error": "access_denied"}
Processing by Users::OmniauthCallbacksController#failure as HTML
  Parameters: {"state"=>"9d2a54295090a659aa20c695e6a9d75005596963efeb27bc", "code"=>"ckkqWoTXmXRcC4GIoMvwekRbrZt9qU"}
Redirected to http://lvh.me:3000/
Completed 302 Found in 0ms (ActiveRecord: 0.0ms)


Started GET "/" for 127.0.0.1 at 2016-08-03 17:38:41 -0600
Processing by HomeController#index as HTML
  Rendered home/index.html.erb within layouts/application (0.3ms)
  Rendered shared/_meta_tags.html.erb (13.8ms)
  Rendered shared/_header.html.erb (0.4ms)
  Rendered shared/_flashes.html.erb (0.1ms)
  Rendered shared/_footer.html.erb (0.1ms)
Completed 200 OK in 20ms (Views: 20.3ms | ActiveRecord: 0.0ms)
```

I've tossed in pry statements and the only one that catches is in the standard "failure" method.

```
app/controllers/users/omniauth_callbacks_controller.rb

class Users::OmniauthCallbacksController < Devise::OmniauthCallbacksController
  def drchrono
    # You need to implement the method below in your model (e.g. app/models/user.rb)
    @user = User.from_omniauth(request.env["omniauth.auth"])
    
    if @user.persisted?
      sign_in_and_redirect @user, :event => :authentication #this will throw if @user is not activated
      set_flash_message(:notice, :success, :kind => "drchrono") if is_navigational_format?
    else
      session["devise.drchrono_data"] = request.env["omniauth.auth"]
      redirect_to new_user_registration_url
    end
  end

  def failure
    redirect_to root_path
  end
end
```

routes.rb

```
devise_for :users, controllers: { omniauth_callbacks: "users/omniauth_callbacks" }
```

link on my devise session new.html.erb

```
<%= link_to "Sign in with Dr Chrono", user_drchrono_omniauth_authorize_path %>
```

config/initializers/devise.rb

```
config.omniauth :drchrono, Rails.application.secrets.drchrono_client_id,
                             Rails.application.secrets.drchrono_client_secret,
                             provider_ignores_state: true
```

It redirects back to my app just fine. I'm able to watch it in console, as well as ngrok. It does send the code, but it seems to not even make it into the drchrono method, so I'm not sure. Maybe it's something wrong with the way Omniauth handles the response from DrChrono.

Any ideas on this are welcome. Thanks!

**Comment by johnnadeau** [comment]
Thanks for writing this up @ShepBook. I've done some quick poking around this morning in my own app and I can't reproduce. Our setup is very similar as far as the basics of using omniauth/devise for authentication goes.

One thing I'm curious about is what version of the drchrono API are you using? My app is still on v2014_08. I haven't yet moved or tested v2015_08 in my app or with this gem. I'm wondering if it's just some incompatibility there.

**Comment by ShepBook** [comment]
Unfortunately I no longer work for the company that I was attempting to integrate Dr Chrono with. I'm going to go ahead and close this. Thanks for your time.

**Comment by johnnadeau** [comment]
I know this is closed, but I'm getting the same thing when I work with https://github.com/PracticallyGreen/omniauth-test-harness and this gem. I'll work to fix that in #3

**Comment by johnnadeau** [comment]
I actually found the issue here, it stems from a discussion on https://github.com/intridea/omniauth-oauth2/issues/81

What it'll result in is me locking the `omniauth-ouauth2` dependency to 1.3.1 as the issue comes up in 1.4. I'll work towards making it compatible with 1.4 in a future change.
